x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/integration/MlRestTestStateCleaner.java

```diff
@@ -91,4 +92,20 @@ private void deleteAllJobs() throws IOException {
             adminClient.performRequest(new Request("DELETE", "/_ml/anomaly_detectors/" + jobId));
         }
     }
+
+    private void deleteAllDataFrameAnalytics() throws IOException {
+        final Request analyticsRequest = new Request("GET", "/_ml/data_frame/analytics?size=10000");
+        analyticsRequest.addParameter("filter_path", "data_frame_analytics");
+        final Response analyticsResponse = adminClient.performRequest(analyticsRequest);
+        List<Map<String, Object>> analytics = (List<Map<String, Object>>) XContentMapValues.extractValue(
+            "data_frame_analytics", ESRestTestCase.entityAsMap(analyticsResponse));
+        if (analytics == null) {
+            return;
+        }
+
+        for (Map<String, Object> config : analytics) {
+            String id = (String) config.get("id");
+            adminClient.performRequest(new Request("DELETE", "/_ml/data_frame/analytics/" + id));
```

Comment:
This isn't needed to delete the configs because the base class will delete the index they're stored in, so they'll all get wiped between tests.

The only thing this is doing is possibly triggering the error that delete throws when you try to delete an analytics process that's running.  If that's the reason this method exists then I think it should be renamed to reflect that.

But maybe `ESRestTestCase.waitForPendingTasks()` already does that check?  In which case there is no need to have this method at all.

(As an aside, the anomaly detector and datafeed methods in this class will no longer need to call delete once we're sure every config is in an index, and then those methods can be renamed.  But that's for a different PR.)